### DISCUSSION TITLE
Player Driven Economy

### DISCUSSION THREAD

**Discussion by LuxPoseidon** [EveronLife/EveronLife#126] [Ideas]
Player Driven Economy would be a cool addition to bring into the trader system, notice 

> https://github.com/EveronLife/EveronLife/issues/8#issuecomment-1146648980

Here you see the #/X that X could be optionally changed to only be able to sell as much as is bought by the trader.

I imagine the prices would adjust based on scarcity to a max / min value per item.

_Originally posted by @LuxPoseidon in https://github.com/EveronLife/EveronLife/issues/8#issuecomment-1155130143_

This system would be an option for servers to enable, they would have further options like:
Max / Min Sell or Buy Prices per item
Persistent Economy (eco persists past restart or does it reset)
Restart Restock (does the server restock supplies on restart)
Restock Amount (adjust how much items restock by after a restart)

**Comment by PackJC** [comment]
Arma 2 life and Arma 3 life has a concenpt of this that should be mirrored.

**Comment by jasper-one** [comment]
I think an exponential Supply & Demand curve would be much better than simply restricting the number of units bought and having arbitrarily fixed min & max prices. 

Somethling like this:
![image](https://user-images.githubusercontent.com/106525617/173774952-26a4680b-b1f3-45e6-b6c6-507573d98592.png)
P = Price |  Q = Quantity

This would make items unrealistically expensive or worthless depending on the amount of demand and it could probably be set up in a way that servers could meet a good equilibrium. 

I'd be happy to help on the math side of things, once we get to the stage of actually doing something like this.

**Reply by LuxPoseidon** [reply]
Definitely this, I just think there should be customizable limits to it that way an apple isn't free or $1,000,000.
